src/queue.c

```diff
@@ -5750,3 +5764,15 @@ _dispatch_context_cleanup(void *ctxt)
 	DISPATCH_INTERNAL_CRASH(ctxt,
 			"Premature thread exit while a dispatch context is set");
 }
+
+#if !DISPATCH_COCOA_COMPAT
+// FIXME:  Need real implementation of this + related functions for Linux
+DISPATCH_NOINLINE
+void
+_dispatch_queue_class_override_drainer(dispatch_queue_t dq,
```

Comment:
the reason is that `_dispatch_queue_class_override_drainer()` only makes sense if you have QoS support which linux doesn't so there's no point having the code in the first place, as you can see _dispatch_queue_class_override_drainer() is not reachable because it is itself under `#if HAVE_PTHREAD_WORKQUEUE_QOS`